src/gdal_read_stream.cpp

```diff
@@ -0,0 +1,133 @@
+
+#include <ogrsf_frmts.h>
+
+#define RCPP_DEFAULT_INCLUDE_CALL false
+#include "Rcpp.h"
+
+#if (GDAL_VERSION_MAJOR >= 3 && GDAL_VERSION_MINOR >= 6)
```

Comment:
would break for GDAL 4.0:

```suggestion
#if GDAL_VERSION_NUM >= GDAL_COMPUTE_VERSION(3,6,0)
```